"Remove braces from single expression" false positive for "unsafe" block

```rust
let slice = &[1];
let start = slice.as_ptr();
let end = unsafe { start.offset(1) };
```
Put your cursor on the `unsafe` keyword -- there is a suggestion to "Remove braces from single expression". But I apply the quickfix, I will get a compile error `error[E0133]: call to unsafe function requires unsafe function or block`.

--- a/src/main/kotlin/org/rust/ide/intentions/UnwrapSingleExprIntention.kt
+++ b/src/main/kotlin/org/rust/ide/intentions/UnwrapSingleExprIntention.kt
@@ -19,6 +19,7 @@ class UnwrapSingleExprIntention : RsElementBaseIntentionAction<RsBlockExpr>() {
 
     override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): RsBlockExpr? {
         val blockExpr = element.parentOfType<RsBlockExpr>() ?: return null
+        if (blockExpr.unsafe != null) return null
         val block = blockExpr.block
 
         return if (block.expr != null && block.lbrace.getNextNonCommentSibling() == block.expr)

--- a/src/test/kotlin/org/rust/ide/intentions/UnwrapSingleExprIntentionTest.kt
+++ b/src/test/kotlin/org/rust/ide/intentions/UnwrapSingleExprIntentionTest.kt
@@ -92,6 +92,14 @@ class UnwrapSingleExprIntentionTest : RsTestBase() {
         """
     )
 
+    fun testUnavailableUnwrapBracesUnsafe() = doUnavailableTest(
+        """
+        fn main() {
+            let wellThen = unsafe<caret> { magic() };
+        }
+        """
+    )
+
     private fun doAvailableTest(@Language("Rust") before: String, @Language("Rust") after: String) {
         myFixture.configureByText(RsFileType, before)
         myFixture.launchAction(UnwrapSingleExprIntention())

